be/src/runtime/runtime_state.cpp

```diff
@@ -158,7 +154,7 @@ Status RuntimeState::init(const TUniqueId& fragment_instance_id, const TQueryOpt
     }
 
     // Register with the thread mgr
-    if (exec_env != nullptr) {
+    if (exec_env != nullptr && !enable_pipeline_engine()) {
```

Comment:
Please add some comments here to explain why _resource_pool is not needed.

Reply:
Resource pool is used to limit the threads used by a query, but pipeline engine does not use these threads.